src/Data/YAML.hs

```diff
@@ -473,11 +473,11 @@ instance FromYAML Word32 where parseYAML = parseInt "Word32"
 instance FromYAML Word64 where parseYAML = parseInt "Word64"
 
 
-instance FromYAML Double where
+instance FromYAML Scientific where
   parseYAML = withFloat "!!float" pure
```

Comment:
What's the rationale for removing `FromYAML` and `ToYAML Double`?

Reply:
I guess there's an argument for going either way.  My thought was that it's mainly there for the vales actually supported as scalar primitives, and since this PR changes the scalar numeric primitive from Double to Scientific, that seemed to be the obvious route to go.